diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5e929c..be8112f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,7 +75,7 @@ option(AOTRITON_INHERIT_SYSTEM_SITE_TRITON "Use system site packages and verify
 # No plan to support network based pip install
 set(AOTRITON_USE_LOCAL_TRITON_WHEEL "" CACHE STRING "Substitute install from third_party/triton with install from local pip wheel package.")
 set(AOTRITON_GPU_BUILD_TIMEOUT "8.0" CACHE STRING "GPU kernel compiler times out after X minutes. 0 for indefinite. Highly recommended if AOTRITON_BUILD_FOR_TUNING=On.")
-set(AOTRITON_TARGET_ARCH "gfx90a;gfx942;gfx950;gfx1100;gfx1101;gfx1102;gfx1151;gfx1150;gfx1201;gfx1200" CACHE STRING "Target GPU Architecture. Select all GPUs within the given list")
+set(AOTRITON_TARGET_ARCH "gfx90a;gfx942;gfx950;gfx1030;gfx1100;gfx1101;gfx1102;gfx1151;gfx1150;gfx1201;gfx1200" CACHE STRING "Target GPU Architecture. Select all GPUs within the given list")
 set(TARGET_GPUS "OBSOLETE" CACHE STRING "OBSOLETE. To select only one GPU, use AOTRITON_TARGET_ARCH or AOTRITON_OVERRIDE_TARGET_GPUS.")
 set(AOTRITON_OVERRIDE_TARGET_GPUS "" CACHE STRING "Override AOTRITON_TARGET_ARCH, and only build for GPUs within this list.")
 
diff --git a/include/aotriton/util.h b/include/aotriton/util.h
index 4ed494f..3de0d7c 100644
--- a/include/aotriton/util.h
+++ b/include/aotriton/util.h
@@ -56,6 +56,7 @@ enum AOTRITON_API Gpu : uint64_t {
   GPU_AMD_ARCH_GFX942_MOD0  = TRICAT(GpuVendor::kAMD,  0x942, 0),
   GPU_AMD_ARCH_GFX942_MOD1  = TRICAT(GpuVendor::kAMD,  0x942, 1),
   GPU_AMD_ARCH_GFX942_MOD2  = TRICAT(GpuVendor::kAMD,  0x942, 2),
+  GPU_AMD_ARCH_GFX1030_MOD0 = TRICAT(GpuVendor::kAMD, 0x1030, 0),
   GPU_AMD_ARCH_GFX1100_MOD0 = TRICAT(GpuVendor::kAMD, 0x1100, 0),
   GPU_AMD_ARCH_GFX1101_MOD0 = TRICAT(GpuVendor::kAMD, 0x1101, 0),
   GPU_AMD_ARCH_GFX1102_MOD0 = TRICAT(GpuVendor::kAMD, 0x1102, 0),
diff --git a/test/_common_test.py b/test/_common_test.py
index 8a16489..043db06 100644
--- a/test/_common_test.py
+++ b/test/_common_test.py
@@ -379,7 +379,8 @@ class SdpaContext(object):
                 key_fudge_factor = 500.0 if self.is_hdim_NPOT_optimized else 340.0
                 bias_fudge_factor = 45.0 if self.is_hdim_NPOT_optimized else 36.0
             # Navi31 needs larger factors
-            if 'gfx1100' in torch.cuda.get_device_properties(0).gcnArchName:
+            if 'gfx1100' in torch.cuda.get_device_properties(0).gcnArchName or \
+                'gfx1030' in torch.cuda.get_device_properties(0).gcnArchName:
                 query_fudge_factor = max(query_fudge_factor, 768.0 if p.dropout_p > 0.0 else 320.0)
         if AOTRITON_TORCH_ONLY_USE_CPU:
             query_fudge_factor = 128.0
diff --git a/tritonsrc/attn_torch_function.py b/tritonsrc/attn_torch_function.py
index 0b7f7cd..a38cefa 100644
--- a/tritonsrc/attn_torch_function.py
+++ b/tritonsrc/attn_torch_function.py
@@ -34,7 +34,7 @@ def evaluate_gfx_arch_within(arch_list):
     return any(arch in gcn_arch_name for arch in arch_list)
 
 def is_rdna():
-    return evaluate_gfx_arch_within(['gfx1100', 'gfx1101', 'gfx1200', 'gfx1201', 'gfx1030', 'gfx1150', 'gfx1151'])
+    return evaluate_gfx_arch_within(['gfx1030', 'gfx1100', 'gfx1101', 'gfx1200', 'gfx1201', 'gfx1030', 'gfx1150', 'gfx1151'])
 
 IS_RDNA = is_rdna()
 
diff --git a/v2python/compile.py b/v2python/compile.py
index 12a4a84..f409a2c 100644
--- a/v2python/compile.py
+++ b/v2python/compile.py
@@ -15,7 +15,7 @@ import json
 from triton.backends.compiler import GPUTarget
 
 KNOWN_TARGETS_MI = ['gfx90a', 'gfx942', 'gfx950']
-KNOWN_TARGETS_Navi = ['gfx1100', 'gfx1101', 'gfx1151', 'gfx1201']
+KNOWN_TARGETS_Navi = ['gfx1030', 'gfx1100', 'gfx1101', 'gfx1151', 'gfx1201']
 
 KNOWN_TARGETS = {
   arch : GPUTarget('hip', arch, 64) for arch in KNOWN_TARGETS_MI
diff --git a/v2python/gpu_targets.py b/v2python/gpu_targets.py
index 622def3..485237c 100644
--- a/v2python/gpu_targets.py
+++ b/v2python/gpu_targets.py
@@ -9,6 +9,7 @@ AOTRITON_SUPPORTED_GPUS = (
     # 'gfx942_mod1',
     # 'gfx942_mod2',
     'gfx950_mod0',
+    'gfx1030_mod0',
     'gfx1100_mod0',
     'gfx1151_mod0',
     'gfx1201_mod0',
@@ -24,6 +25,7 @@ AOTRITON_ARCH_WARPSIZE = {
     'gfx90a'     : 64,
     'gfx942'     : 64,
     'gfx950'     : 64,
+    'gfx1030'    : 32,
     'gfx1100'    : 32,
     'gfx1151'    : 32,
     'gfx1201'    : 32,
diff --git a/v2python/tuning_lut.py b/v2python/tuning_lut.py
index 723b744..d307930 100644
--- a/v2python/tuning_lut.py
+++ b/v2python/tuning_lut.py
@@ -15,6 +15,7 @@ SKIPPED_LUT_CHECK = os.getenv('AOTRITON_SKIP_LUT_CHECK', default='').split(',')
 ARCH_TO_DIRECTORY = {
     'gfx90a'  : 'amd-gfx90a',
     'gfx942' : 'amd-gfx942',
+    'gfx1030' : 'amd-gfx103x',
     'gfx1100' : 'amd-gfx110x',
     'gfx1101' : 'amd-gfx110x',
     'gfx1151' : 'amd-gfx115x',
diff --git a/v2src/util.cc b/v2src/util.cc
index a66488e..034f9da 100644
--- a/v2src/util.cc
+++ b/v2src/util.cc
@@ -48,6 +48,7 @@ private:
 std::unordered_map<std::string, GpuClassifier> LazyGpu::string_to_classifier = {
   { "gfx90a", DummyClassifier<GPU_AMD_ARCH_GFX90A_MOD0 >() },
   { "gfx942", DummyClassifier<GPU_AMD_ARCH_GFX942_MOD0 >() },
+  {"gfx1030", DummyClassifier<GPU_AMD_ARCH_GFX1030_MOD0>() },
   {"gfx1100", DummyClassifier<GPU_AMD_ARCH_GFX1100_MOD0>() },
   {"gfx1101", DummyClassifier<GPU_AMD_ARCH_GFX1101_MOD0>() },
   {"gfx1151", DummyClassifier<GPU_AMD_ARCH_GFX1151_MOD0>() },
diff --git a/v3python/compile.py b/v3python/compile.py
index 61477ac..3d13d6d 100644
--- a/v3python/compile.py
+++ b/v3python/compile.py
@@ -15,7 +15,7 @@ import json
 from triton.backends.compiler import GPUTarget
 
 KNOWN_TARGETS_64 = ['gfx90a', 'gfx942', 'gfx950']
-KNOWN_TARGETS_32 = ['gfx1100', 'gfx1101', 'gfx1102', 'gfx1201', 'gfx1200', 'gfx1151', 'gfx1150', 'gfx1250']
+KNOWN_TARGETS_32 = ['gfx1030', 'gfx1100', 'gfx1101', 'gfx1102', 'gfx1201', 'gfx1200', 'gfx1151', 'gfx1150', 'gfx1250']
 
 KNOWN_TARGETS = {
   arch : GPUTarget('hip', arch, 64) for arch in KNOWN_TARGETS_64
diff --git a/v3python/gpu_targets.py b/v3python/gpu_targets.py
index 8492757..aa5b6a0 100644
--- a/v3python/gpu_targets.py
+++ b/v3python/gpu_targets.py
@@ -9,6 +9,7 @@ AOTRITON_SUPPORTED_GPUS = (
     # 'gfx942_mod1',
     # 'gfx942_mod2',
     'gfx950_mod0',
+    'gfx1030_mod0',
     'gfx1100_mod0',
     'gfx1101_mod0',
     'gfx1102_mod0',
@@ -34,6 +35,7 @@ AOTRITON_ARCH_TO_PACK = {
     'gfx90a'    : 'gfx90a',
     'gfx942'    : 'gfx942',
     'gfx950'    : 'gfx950',
+    'gfx1030'   : 'gfx103x',
     'gfx1100'   : 'gfx11xx',
     'gfx1101'   : 'gfx11xx',
     'gfx1102'   : 'gfx11xx',
@@ -52,6 +54,7 @@ AOTRITON_ARCH_WARPSIZE = {
     'gfx90a'     : 64,
     'gfx942'     : 64,
     'gfx950'     : 64,
+    'gfx1030'    : 32,
     'gfx1100'    : 32,
     'gfx1101'    : 32,
     'gfx1102'    : 32,
@@ -66,6 +69,7 @@ AOTRITON_ARCH_PRODUCTION_LINE = {
     'gfx90a'     : 'CDNA',
     'gfx942'     : 'CDNA',
     'gfx950'     : 'CDNA',
+    'gfx1030'    : 'RDNA',
     'gfx1100'    : 'RDNA',
     'gfx1101'    : 'RDNA',
     'gfx1102'    : 'RDNA',
diff --git a/v3src/util.cc b/v3src/util.cc
index f70804e..3a8c56a 100644
--- a/v3src/util.cc
+++ b/v3src/util.cc
@@ -48,6 +48,7 @@ private:
 std::unordered_map<std::string, GpuClassifier> LazyGpu::string_to_classifier = {
   { "gfx90a", DummyClassifier<GPU_AMD_ARCH_GFX90A_MOD0 >() },
   { "gfx942", DummyClassifier<GPU_AMD_ARCH_GFX942_MOD0 >() },
+  {"gfx1030", DummyClassifier<GPU_AMD_ARCH_GFX1030_MOD0>() },
   {"gfx1100", DummyClassifier<GPU_AMD_ARCH_GFX1100_MOD0>() },
   {"gfx1101", DummyClassifier<GPU_AMD_ARCH_GFX1101_MOD0>() },
   {"gfx1102", DummyClassifier<GPU_AMD_ARCH_GFX1102_MOD0>() },
